src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedLocal.cs

```diff
@@ -33,13 +35,13 @@ internal sealed class SynthesizedLocal : LocalSymbol
             MethodSymbol containingMethodOpt,
             TypeWithAnnotations type,
             SynthesizedLocalKind kind,
-            SyntaxNode syntaxOpt = null,
+            SyntaxNode? syntaxOpt = null,
             bool isPinned = false,
             RefKind refKind = RefKind.None
 #if DEBUG
             ,
             [CallerLineNumber]int createdAtLineNumber = 0,
-            [CallerFilePath]string createdAtFilePath = null
+            [CallerFilePath]string createdAtFilePath = null!
```

Comment:
>string createdAtFilePath = null! [](start = 28, length = 32)

why not use `string?` instead?

Reply:
The default value `null` isn't what gets used, so this seemed more accurate. Ideally the compiler would produce a warning if the default value gets used, or if the file path was not available and `null` was substituted.